File: applications/paper/all/mapc2m_cylinder.f
      subroutine mapc2m_cylinder2(xc,yc,xp,yp,zp)
      implicit none

      double precision xc,yc,xp,yp,zp

      double precision alpha, beta
      common /cylinder_comm/ alpha, beta

      double precision r1, R, theta, phi

      call map_comp2cylinder(xc,yc,theta,phi)      

      r1 = alpha*(1 + beta*sin(theta))
      R = 1 + r1*cos(phi)

      xp = R*cos(theta)
      yp = R*sin(theta)
      zp = r1*sin(phi)

      end

File: applications/paper/transport_cylinder/mapc2m_cylinder.m
function [xp,yp,zp] = mapc2m_cylinder(xc,yc,R,H)


[~,~,~, R, H,~] = read_vars();

pi2 = 2*pi;

theta = pi2*xc;
z = yc;

xp = R.*cos(theta);
yp = R.*sin(theta);
zp = H*z;

% xp = R.*cos(theta);
% zp = R.*sin(theta);
% yp = H*z;
% zp(zp < 0) = nan;



end